nixos/modules/programs/gnupg.nix

```diff
@@ -17,8 +17,6 @@ let
     || xserverCfg.desktopManager.plasma5.enable
     || xserverCfg.desktopManager.deepin.enable then
       "qt"
-    else if xserverCfg.desktopManager.xfce.enable then
-      "gtk2"
```

Comment:
What's the reason to drop this rule?
I don't think xfce4 users want a terminal based pinentry, or do they?

Reply:
Taking a step back, it seems that this is more of a task for the individual desktop environment modules, to add the right pinentry flavor to `PATH` when initializing their session. That way it works right even if you have multiple installed desktop environments.
Also it solves the architectural flaw of having to enumerate all desktop environments here.
That doesn't have to be solved in this PR, but seems to make like a good next step.

Reply:
> What's the reason to drop this rule? I don't think xfce4 users want a terminal based pinentry, or do they?

They might, xfce users use ttys or go into a non-graphical environment from time to time.

Reply:
I think we can drop this change after 3cdf1705f5374d82d7fa208be92b86e6320ec3f8 which fixes the bug I avoided here.

Reply:
Just felt like doing this in this PR. I hope this is the last step to make this perfect and we don't increase the scope further.